src/System.Net.Sockets/tests/FunctionalTests/KeepAliveTest.netcoreapp.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class KeepAliveTest : IDisposable
+    {
+        private const bool enabled = true;
+        private const int retryCount = 60;
+        private const int time = 5;
+        private const int interval = 2;
+
+        private readonly Socket socket;
```

Comment:
I don't think this should be member, but rather creates in each test. This way the tests share state and it matters in which order they are executed.

Reply:
I agree with Karel, though not for the same reason (xunit creates a new test class for each test invocation, so there isn't actually state sharing or ordering issues here). It's easier to reason about each test when it locally contains/initializes all its state. The test should also be disposing of the socket.

Reply:
Indeed that was a problem of NUnit only:
https://xunit.github.io/docs/why-did-we-build-xunit-1.0